spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -66,7 +164,7 @@ private[connector] class CassandraRunner(val configTemplate: String) extends Emb
   val confDir = mkdir(new File(tempDir, "conf"))
   val confFile = new File(confDir, "cassandra.yaml")
 
-  private val properties = Map("cassandra_dir" -> workDir.toString)
+  private val properties = if(props != null) Map("cassandra_dir" -> workDir.toString) ++ props else Map("cassandra_dir" -> workDir.toString)
```

Comment:
Is it ever possible for `props` to be `null`? If not, you can remove this check. Instead you can add `assert(props != null)` or `require`